### ISSUE TITLE
Provide better errors than "Unexpectedly disconnected from server"

### ISSUE THREAD

**Issue by astei** [VelocityPowered/Velocity#233]
Simple: this one vague error runs the gamut from the user not correctly setting up player information forwarding to issues with Forge.

**Comment by astei** [comment]
I think I've found the common thread. This boils down to an implementation difference between Spigot and Sponge.

In Spigot, the server address field is widened to 32KiB to provide room for the handshaking data. If handshaking data is provided by the proxy to a server without BungeeCord forwarding enabled, it seems to simply ignore it and proceed as usual.

Sponge is different. If the `bungeecord` module is on and `ip-forwarding` is enabled, it increases the limit to the same 32KiB as Spigot does. But if it's not, it uses the default 255-character limit from the vanilla server.

I think the culprit is obvious and the Spigot behavior is broken and insane. But the way that Sponge reacts is also problematic, in that it's not particularly user-friendly (are they doing this for security?)

**Comment by TheMaddog91** [comment]
I've got the latest version of Velocity. I've followed the instructions, i've a lobby server, i've a sevtech & continuum server all on the same machine. However when a play joins the lobby, tries to go through a portal that takes them to the server of their choosing or typing /server sevtech nothing happens.

However the logs spew out this;
https://pastebin.com/AQjf8vUt

I've tried Bungeecord & i've tried waterfall, due to lack of documentation i've decided to try use Velocity. Both Bungeecord and Waterfall will only work if SpongeForge is install and I don't want it on my Continuum & Sevtech modded servers. It breaks too many things.
